src/plugins/atomic/AtomicWidget.cpp

```diff
@@ -0,0 +1,273 @@
+    if(!ui->btn_clearnet->isChecked() && conf()->get(Config::proxy).toInt() != Config::Proxy::None) {
+        arguments << "--tor-socks5-port";
+        arguments << conf()->get(Config::socks5Port).toString();
+    } else if (ui->btn_clearnet->isChecked()) {
+        arguments << "--tor-socks5-port";
+        arguments << "0";
+    }
+    arguments << "--rendezvous-point";
+    arguments << rendezvous;
+    qDebug() << "rendezvous point: " + rendezvous;
+    auto *swap = new QProcess();
+    procList->append(QSharedPointer<QProcess>(swap));
+    swap->setReadChannel(QProcess::StandardError);
+    connect(swap, &QProcess::finished, this, [this, swap]{
+        QJsonDocument parsedLine;
+        QJsonParseError parseError;
+        QList<QSharedPointer<OfferEntry>> list;
+        auto output = QString::fromLocal8Bit(swap->readAllStandardError());
+        auto lines = output.split(QRegularExpression("[\r\n]"),Qt::SkipEmptyParts);
+
+
+        for(const auto& line : lines){
+            qDebug() << line;
+            if(line.contains("status")){
+                parsedLine = QJsonDocument::fromJson(line.toLocal8Bit(), &parseError );
+                if (parsedLine["fields"]["status"].toString().contains("Online")){
+                    bool skip = false;
+                    auto  entry = new OfferEntry(parsedLine["fields"]["price"].toString().split( ' ')[0].toDouble(),parsedLine["fields"]["min_quantity"].toString().split(' ')[0].toDouble(),parsedLine["fields"]["max_quantity"].toString().split(' ')[0].toDouble(), parsedLine["fields"]["address"].toString());
+                    for(const auto& post : *offerList){
+                        if(entry->max == 0 || std::equal(entry->address.begin(), entry->address.end(),post->address.begin(),post->address.end()))
+                            skip = true;
+                    }
+                    if (!skip) {
+                        ui->meta_label->setText("Updated offer book");
+                        offerList->append(QSharedPointer<OfferEntry>(entry));
+                    }
+                }
+            } else if (line.contains("GLIBC_")){
+                QMessageBox::critical(this, "GLIBC outdated", "Upgrade your GLIBC to at least 2.32 to use this tool");
+                clean();
+            }
+        }
+        swap->close();
+        o_model->updateOffers(*offerList);
+        return list;
+    });
+    swap->start(conf()->get(Config::swapPath).toString(), arguments);
+
+
+
+}
+
+AtomicWidget::~AtomicWidget() {
+    qDebug()<< "Exiting widget!!";
+    delete swapDialog;
+    delete o_model;
+    delete offerList;
+    clean();
+    delete procList;
+}
+
+void AtomicWidget::clean() {
+    for (const auto& proc : *procList){
+        proc->kill();
+    }
+    auto cleanWallet =  new QProcess;
+    auto cleanSwap = new QProcess;
+    if(conf()->get(Config::operatingSystem)=="WINDOWS"){
+        (cleanWallet)->start("tskill", QStringList{"monero-wallet-rpc"});
```

Comment:
The way the swap tool functions it needs to have a monero-wallet-rpc endpoint running. This kill is to fix when feather wallet is closed while the swap tool is running as the swap tool doesn't close the wallet rpc when terminated during execution or free it's lock. Without killing both processes relaunching feather will prevent the swap tool from being able to be executed.

Reply:
How are users expected to set up the `monero-wallet-rpc` endpoint?

Reply:
Sorry I didn't explain fully. The swap binary will start `monero-wallet-rpc` if it isn't supplied a port to connect to and also pull and download it if it isn't in the path. The issue is that because the binary won't kill the child process sometimes if terminated mid swap we need to manually kill it to prevent the lock.